editor/import/editor_scene_importer_gltf.cpp

```diff
@@ -1404,53 +1404,81 @@ Error EditorSceneImporterGLTF::_parse_skins(GLTFState &state) {
 	if (!state.json.has("skins"))
 		return OK;
 
-	Array skins = state.json["skins"];
-	for (int i = 0; i < skins.size(); i++) {
+	// Map of joints (nodes) to their appropriate skins
+	Map<int, int> joints_to_skins;
+
+	// Map of copies of a skin to its original instance
+	Map<int, int> skincopy_to_skin;
 
-		Dictionary d = skins[i];
+	const Array &skins = state.json["skins"];
+	for (int i = 0; i < skins.size(); i++) {
 
-		GLTFSkin skin;
+		const Dictionary &d = skins[i];
 
 		ERR_FAIL_COND_V(!d.has("joints"), ERR_PARSE_ERROR);
 
-		Array joints = d["joints"];
+		const Array &joints = d["joints"];
 		Vector<Transform> bind_matrices;
 
 		if (d.has("inverseBindMatrices")) {
 			bind_matrices = _decode_accessor_as_xform(state, d["inverseBindMatrices"], false);
 			ERR_FAIL_COND_V(bind_matrices.size() != joints.size(), ERR_PARSE_ERROR);
 		}
 
+		int duplicate_skin_id = -1;
+
+		// Find any nodes we possibly missed. Ideally, if this is a skin_copy the joints_to_add
+		// should always be empty, otherwise there is an exporter problem.
+		// Thankfully, we can handle it by adding them to the real skin
+		Vector<int> joints_to_add;
 		for (int j = 0; j < joints.size(); j++) {
-			int index = joints[j];
-			ERR_FAIL_INDEX_V(index, state.nodes.size(), ERR_PARSE_ERROR);
+			const int node_index = joints[j];
+
+			if (Map<int, int>::Element *joint_to_skin = joints_to_skins.find(node_index)) {
+				duplicate_skin_id = joint_to_skin->value();
+				continue;
+			}
+
+			joints_to_add.push_back(node_index);
+		}
+
+		GLTFSkin new_skin;
+
+		if (d.has("name")) {
+			new_skin.name = d["name"];
+		}
+
+		// The target skin for these nodes
+		int target_skin_id = duplicate_skin_id == -1 ? i : duplicate_skin_id;
+		GLTFSkin *target_skin = duplicate_skin_id == -1 ? &new_skin : &state.skins[duplicate_skin_id];
```

Comment:
&state.skins[duplicate_skin_id] crashes with an invalid index.

Also a style criticism is to use if and else because this is getting complicated.